src/go/cmd/http-relay-server/broker.go

```diff
@@ -127,6 +132,7 @@ func (r *broker) RelayRequest(server string, request *pb.HttpRequest) (<-chan *p
 		r.req[server] = make(chan *pb.HttpRequest)
 	}
 	if r.resp[id] != nil {
+		r.m.Unlock()
```

Comment:
wow, could you send this as a separate PR (you can send a PR right from the github webpage).

Reply:
Fixed separately in #173